File: articles/personality-character-and-individual-differences-in-moral-life/go/moral_response_profile.go
// Synthetic moral response profile example.
// Educational scaffold only.
//
// Run:
//   go run go/moral_response_profile.go

package main

import "fmt"

func main() {
	broadPersonality := 0.4
	characterStrength := 0.8
	moralIdentity := 0.7
	empathy := 0.6
	selfRegulation := 0.5
	humility := 0.4
	selfInterestBias := 0.3

	profile := 0.20*broadPersonality +
		0.30*characterStrength +
		0.35*moralIdentity +
		0.25*empathy +
		0.20*selfRegulation +
		0.15*humility -
		0.35*selfInterestBias

	fmt.Printf("Synthetic moral response profile: %.3f\n", profile)
}

File: articles/personality-character-and-individual-differences-in-moral-life/cpp/moral_response_profile.cpp
/*
Synthetic moral response profile example.
Educational scaffold only.

Compile:
  c++ -std=c++17 cpp/moral_response_profile.cpp -o outputs/tables/moral_response_profile_cpp
Run:
  ./outputs/tables/moral_response_profile_cpp
*/

#include <iostream>

int main() {
    double broadPersonality = 0.4;
    double characterStrength = 0.8;
    double moralIdentity = 0.7;
    double empathy = 0.6;
    double selfRegulation = 0.5;
    double humility = 0.4;
    double selfInterestBias = 0.3;

    double profile =
        0.20 * broadPersonality +
        0.30 * characterStrength +
        0.35 * moralIdentity +
        0.25 * empathy +
        0.20 * selfRegulation +
        0.15 * humility -
        0.35 * selfInterestBias;

    std::cout << "Synthetic moral response profile: "
              << profile << std::endl;

    return 0;
}
